storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/ConstraintSolverStrategy.java

```diff
@@ -290,15 +301,15 @@ public SchedulingResult schedule(Cluster cluster, TopologyDetails td) {
         }
 
         //early detection/early fail
-        if (!checkSchedulingFeasibility()) {
+        if (!checkSchedulingFeasibility(daemonMaxStateSearch)) {
```

Comment:
nit: rename it to maxStateSearch (it's just the maxStateSearch that has been determined from the config for this topology)

Reply:
making this change. this is actually a bug fix as the previous code just used MAX_STATE_SEARCH